File: signup/missing-fields.java
package com.example.tests;

import java.util.regex.Pattern;
import java.util.concurrent.TimeUnit;
import org.junit.*;
import static org.junit.Assert.*;
import static org.hamcrest.CoreMatchers.*;
import org.openqa.selenium.*;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

public class New {
  private WebDriver driver;
  private String baseUrl;
  private boolean acceptNextAlert = true;
  private StringBuffer verificationErrors = new StringBuffer();

  @Before
  public void setUp() throws Exception {
    driver = new FirefoxDriver();
    baseUrl = "https://sanguinebio.com/";
    driver.manage().timeouts().implicitlyWait(30, TimeUnit.SECONDS);
  }

  @Test
  public void testNew() throws Exception {
    driver.get("https://sanguinebio.com");
    driver.findElement(By.linkText("Sign Up")).click();
    driver.findElement(By.cssSelector("label")).click();
    driver.findElement(By.id("continue-button")).click();
    driver.findElement(By.id("first-name")).clear();
    driver.findElement(By.id("first-name")).sendKeys("test");
    driver.findElement(By.id("last-name")).clear();
    driver.findElement(By.id("last-name")).sendKeys("testing");
    driver.findElement(By.id("signup-submit")).click();
    try {
      assertEquals("Try Again", driver.findElement(By.id("signup-submit")).getText());
    } catch (Error e) {
      verificationErrors.append(e.toString());
    }
  }

  @After
  public void tearDown() throws Exception {
    driver.quit();
    String verificationErrorString = verificationErrors.toString();
    if (!"".equals(verificationErrorString)) {
      fail(verificationErrorString);
    }
  }

  private boolean isElementPresent(By by) {
    try {
      driver.findElement(by);
      return true;
    } catch (NoSuchElementException e) {
      return false;
    }
  }

  private boolean isAlertPresent() {
    try {
      driver.switchTo().alert();
      return true;
    } catch (NoAlertPresentException e) {
      return false;
    }
  }

  private String closeAlertAndGetItsText() {
    try {
      Alert alert = driver.switchTo().alert();
      String alertText = alert.getText();
      if (acceptNextAlert) {
        alert.accept();
      } else {
        alert.dismiss();
      }
      return alertText;
    } finally {
      acceptNextAlert = true;
    }
  }
}

File: signup/missing-fields.js
// verify that signup fails if form is not complete

var fs = require('fs');
var webdriver = require('selenium-webdriver');
var By = webdriver.By;
var until = webdriver.until;

var driver = new webdriver.Builder()
  .forBrowser('firefox')
  .build();

driver.get("https://sanguinebio.com");
driver.findElement(By.className('signup-link')).click();
driver.findElement(By.css("label")).click();
driver.findElement(By.id("continue-button")).click();
driver.findElement(By.id("first-name")).clear();
driver.findElement(By.id("first-name")).sendKeys("test");
driver.findElement(By.id("last-name")).clear();
driver.findElement(By.id("last-name")).sendKeys("testing");
driver.findElement(By.id("signup-submit")).click();

driver.takeScreenshot().then(function(data) {
  writeScreenshot(data, 'missing-fields.png');
});

var textPromise = driver.findElement(By.id("signup-submit")).getText();

textPromise.then(function(text) {
  if (text === "TRY AGAIN") {
    console.log("success");
    process.exit(0);
    return;
  } else {
    console.log("fail");
    process.exit(1);
  }
});

function writeScreenshot(data, name) {
  name = name || 'ss.png';
  var screenshotPath = '/tmp/';
  fs.writeFileSync(screenshotPath + name, data, 'base64');
};
